moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -200,8 +200,9 @@ void plan_execution::PlanExecution::planAndExecuteHelper(ExecutableMotionPlan& p
       if (plan.error_code_.val == moveit_msgs::msg::MoveItErrorCodes::UNABLE_TO_AQUIRE_SENSOR_DATA &&
           opt.replan_delay_ > 0.0)
       {
-        ros::WallDuration d(opt.replan_delay_);
-        d.sleep();
+        auto d = std::chrono::duration<double>(opt.replan_delay_);
+        auto period = std::chrono::milliseconds(int(d.count()*1000));
```

Comment:
This is overly complicated, also `sleep_for` expects nanoseconds. Just cast `d` (better rename to replan_delay_seconds or so) directly to nanoseconds.